File: ts/wam/7064.ts
export const WamGroupHistoryReceiverUserJourney = 7064 as const

export interface GroupHistoryReceiverUserJourneyEvent {
  /** field 1, wire `group_history_messages_count` */
  groupHistoryMessagesCount?: number
  /** field 2, wire `group_history_receiver_action_type` */
  groupHistoryReceiverActionType?: GROUPHISTORYRECEIVERUSERJOURNEYACTIONTYPE
  /** field 3, wire `group_history_receiver_group_id` */
  groupHistoryReceiverGroupId?: string
  /** field 4, wire `is_auto_process` */
  isAutoProcess?: boolean
  /** field 5, wire `message_received_ts` */
  messageReceivedTs?: number
  /** field 6, wire `ui_surface` */
  uiSurface?: TSSURFACE
  /** field 7, wire `unified_session_id` */
  unifiedSessionId?: string
  /** field 8, wire `user_journey_ms` */
  userJourneyMs?: number
  /** field 9, wire `group_history_db_ignored_older_messages` */
  groupHistoryDbIgnoredOlderMessages?: boolean
  /** field 10, wire `message_key_hash` */
  messageKeyHash?: string
  /** field 11, wire `receiver_failure_reason` */
  receiverFailureReason?: string
  /** field 12, wire `group_history_out_window_pins_count` */
  groupHistoryOutWindowPinsCount?: number
  /** field 13, wire `group_history_pins_count` */
  groupHistoryPinsCount?: number
  /** field 14, wire `group_history_uncounted_messages_count` */
  groupHistoryUncountedMessagesCount?: number
}

File: go/wam/7064.go
const WamGroupHistoryReceiverUserJourney = 7064 // channel: regular

type GroupHistoryReceiverUserJourneyEvent struct {
	GroupHistoryMessagesCount int64 `wam:"group_history_messages_count"` // field 1
	GroupHistoryReceiverActionType GROUPHISTORYRECEIVERUSERJOURNEYACTIONTYPE `wam:"group_history_receiver_action_type"` // field 2
	GroupHistoryReceiverGroupId string `wam:"group_history_receiver_group_id"` // field 3
	IsAutoProcess bool `wam:"is_auto_process"` // field 4
	MessageReceivedTs int64 `wam:"message_received_ts"` // field 5
	UiSurface TSSURFACE `wam:"ui_surface"` // field 6
	UnifiedSessionId string `wam:"unified_session_id"` // field 7
	UserJourneyMs int64 `wam:"user_journey_ms"` // field 8
	GroupHistoryDbIgnoredOlderMessages bool `wam:"group_history_db_ignored_older_messages"` // field 9
	MessageKeyHash string `wam:"message_key_hash"` // field 10
	ReceiverFailureReason string `wam:"receiver_failure_reason"` // field 11
	GroupHistoryOutWindowPinsCount int64 `wam:"group_history_out_window_pins_count"` // field 12
	GroupHistoryPinsCount int64 `wam:"group_history_pins_count"` // field 13
	GroupHistoryUncountedMessagesCount int64 `wam:"group_history_uncounted_messages_count"` // field 14
}